packages/client/src/actions/applyAction.ts

```diff
@@ -14,21 +14,77 @@
  * limitations under the License.
  */
 
-import type { ActionDefinition } from "@osdk/api";
 import type {
+  ActionDefinition,
+  ActionParameterDefinition,
+  ObjectActionDataType,
+  ObjectSetActionDataType,
+} from "@osdk/api";
+import type {
+  ActionParam,
   ActionReturnTypeForOptions,
   ApplyActionOptions,
   ApplyBatchActionOptions,
-  OsdkActionParameters,
+  DataValueClientToWire,
+  NOOP,
+  PartialBy,
 } from "@osdk/client.api";
 import type { DataValue } from "@osdk/internal.foundry";
-import { Ontologies, OntologiesV2 } from "@osdk/internal.foundry";
+import { OntologiesV2 } from "@osdk/internal.foundry";
 import type { MinimalClient } from "../MinimalClientContext.js";
 import { addUserAgentAndRequestContextHeaders } from "../util/addUserAgentAndRequestContextHeaders.js";
 import { augmentRequestContext } from "../util/augmentRequestContext.js";
+import type { NullableProps } from "../util/NullableProps.js";
 import { toDataValue } from "../util/toDataValue.js";
 import { ActionValidationError } from "./ActionValidationError.js";
 
+type BaseType<APD extends Pick<ActionParameterDefinition<any, any>, "type">> =
```

Comment:
All these types moved back from @client.api needed a home and this seemed as good as any.